internal/collector/collector.go

```diff
@@ -212,7 +213,8 @@ func (f *Factory) NewCollectorContainer(secretNames []string, clusterID string,
 	}
 	for _, receiverInput := range receiverInputs {
 		collector.VolumeMounts = append(collector.VolumeMounts,
-			v1.VolumeMount{Name: receiverInput, ReadOnly: true, MountPath: path.Join(receiverInputVolumePath, receiverInput)},
+			// make receivers compatible with the next generation conf generation
+			v1.VolumeMount{Name: receiverInput, ReadOnly: true, MountPath: path.Join(receiverInputVolumePath, strings.ReplaceAll("input-" + receiverInput, "-", "_"))},
```

Comment:
Why does the mount path need to change here? input names are unique.  If it is because of the change in the wiring, then we should consider:
* modifying the wiring to use the name (this is the option I prefer)
* use the helper method `helpers.MakeInputID` to craft this method

Reply:
Absolutely right, that was very bad.

I have changed the wiring for both http (which was also failing with the new conf) and syslog receivers to use the name instead of the ID when searching for the certs directory.